src/Ormolu/Printer/Meat/Declaration/Value.hs

```diff
@@ -971,28 +971,56 @@ p_if ::
   (body -> Placement) ->
   -- | Render
   (body -> R ()) ->
+  -- | Annotations
+  EpAnn AnnsIf ->
   -- | If
   LHsExpr GhcPs ->
   -- | Then
   LocatedA body ->
   -- | Else
   LocatedA body ->
   R ()
-p_if placer render if' then' else' = do
+p_if placer render epAnn if' then' else' = do
   txt "if"
   space
   located if' p_hsExpr
   breakpoint
   inci $ do
-    txt "then"
+    locatedToken thenSpan "then"
     space
-    located then' $ \x ->
-      placeHanging (placer x) (render x)
+    placeHangingLocated thenSpan then'
     breakpoint
-    txt "else"
+    locatedToken elseSpan "else"
     space
-    located else' $ \x ->
-      placeHanging (placer x) (render x)
+    placeHangingLocated elseSpan else'
+  where
+    (thenSpan, elseSpan, commentSpans) =
+      case epAnn of
+        EpAnn {anns = AnnsIf {aiThen, aiElse}, comments} ->
+          ( loc' $ epaLocationRealSrcSpan aiThen,
+            loc' $ epaLocationRealSrcSpan aiElse,
+            map (anchor . getLoc) $
+              case comments of
+                EpaComments cs -> cs
+                EpaCommentsBalanced pre post -> pre <> post
+          )
+        EpAnnNotUsed ->
+          (noSrcSpan, noSrcSpan, [])
+
+    locatedToken tokenSpan token =
+      located (L tokenSpan ()) $ \_ -> txt token
```

Comment:
```suggestion
      switchLayout [tokenSpan] (txt token)
```

Reply:
This doesn't work (you will see local test failures) as `switchLayout` doesn't emit comments.

Reply:
Eh, indeed.